src/Proxy.hs

```diff
@@ -507,6 +512,76 @@ getTransactionCostR = withExchangeRate $ \(rate, pv) -> do
                 costResponse $ bakerConfigureEnergyCostWithoutKeys pSize numSignatures
+              "update" -> do
+                invoker <- lookupGetParam "invoker" >>= \case
+                      Nothing -> respond400Error (EMParseError "Missing `invoker` value.") RequestInvalid
+                      Just val -> case addressFromText val of
+                        Left s -> respond400Error (EMParseError $ "Could not parse `invoker` value: " ++ s) RequestInvalid
+                        Right addr -> return $ Just $ AddressAccount addr
+
+                contractIndex <- lookupGetParam "contractIndex" >>= \case
+                      Nothing -> respond400Error (EMParseError "Missing `contractIndex` value.") RequestInvalid
+                      Just val -> case readMaybe $ Text.unpack val of
+                        Nothing -> respond400Error (EMParseError "Could not parse `contractIndex` value.") RequestInvalid
+                        Just index -> return $ ContractIndex index
+
+                contractSubindex <- lookupGetParam "contractSubindex" >>= \case
+                      Nothing -> respond400Error (EMParseError "Missing `contractSubindex` value.") RequestInvalid
+                      Just val -> case readMaybe $ Text.unpack val of
+                        Nothing -> respond400Error (EMParseError "Could not parse `contractSubindex` value.") RequestInvalid
+                        Just index -> return $ ContractSubindex index
+
+                let contract = ContractAddress{..}
+
+                amount <- lookupGetParam "amount" >>= \case
+                      Nothing -> respond400Error (EMParseError "Missing `amount` value.") RequestInvalid
+                      Just val -> case readMaybe $ Text.unpack val of
+                        Nothing -> respond400Error (EMParseError "Could not parse `amount` value.") RequestInvalid
+                        Just a -> return a
+                wasmReceiveName <- lookupGetParam "receiveName" >>= \case
+                      Nothing -> respond400Error (EMParseError "Missing `receiveName` value.") RequestInvalid
+                      Just receiveName -> if Wasm.isValidReceiveName receiveName
+                        then return $ Wasm.ReceiveName receiveName
+                        else respond400Error (EMParseError "Invalid receive name.") RequestInvalid
+                wasmParameter <- lookupGetParam "parameter" >>= \case
+                      Nothing -> respond400Error (EMParseError "Missing `parameter` value.") RequestInvalid
+                      Just parameterText -> case BS16.decode . Text.encodeUtf8 $ parameterText of
+                        Left s -> respond400Error (EMParseError $ "Could not parse `parameter` value: " ++ s) RequestInvalid
+                        Right parameter -> return $ Wasm.Parameter $ BSS.toShort parameter
+
+                let nrg = Energy 30000000 -- 30 million
```

Comment:
I think we should be much more conservative here, and not allow more than 500ms to execute the estimate. So set this to 500_000

Reply:
Done.